lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -671,6 +684,12 @@ def _push_alerts_to_relation_databag(self, relation: Relation) -> None:
 
         alert_rules_as_dict = alert_rules.as_dict()
 
+        if self.extra_alert_rules:
+            if alert_rules_as_dict:
+                alert_rules_as_dict["groups"].extend(self.extra_alert_rules)
+            elif self.extra_alert_rules:
+                alert_rules_as_dict["groups"] = self.extra_alert_rules
```

Comment:
Could this be simplified a bit without breaking existing contracts?
Maybe this would work?
```python
alert_rules_as_dict = {"groups": alert_rules_as_dict.get("groups", []) + self.extra_alert_rules}
```

Reply:
then `alert_rules_as_dict` always will be True, causing next condition in the code to fire and fill relation databag with confusing data